Azure/sonic-buildimage
# Ensure redis client socket is closed and prevent race condition

#### Why I did it
1. Prevent race condition inside get_all_namespaces() which is a not a reentrant function
2. Ensure no FD leak as this fix reverts this [PR9636](https://github.com/Azure/sonic-buildimage/pull/9636)


#### How I did it
1. Store the redis client connection for CONFIG_DB in a local variable 
2. Forcefully close the redis client connection (not required since local connection object get destroyed once out of scope, but just in case)

#### How to verify it
Ran multiple times
1 show interface counters
2. portstat -a

Ensure the redis client socket connection gets closed after the get_all_namespace() returns
Ensure no fd leak 
Ensure no race-condition as observed in Cisco chassis when multiple Xcvrd threads calls get_all_namespace()

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
